src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/21.diff.sql

```diff
@@ -0,0 +1,39 @@
+-- STORED PROCEDURE
+--     RemovePartitionFromResourceChanges_2
+--
+-- DESCRIPTION
+--     Switches a partition from a resource change data table to a staging table.
+--     Then merges two partitions based on a given partition boundary value.
+--     After that, truncates the staging table to purge the old resource change data.
+--
+--     @partitionNumberToSwtichOut
+--         * The partition number to switch out.
+--     @partitionBoundaryToMerge
+--         * The partition boundary value to merge.
+--
+CREATE OR ALTER PROCEDURE dbo.RemovePartitionFromResourceChanges_2
+    @partitionNumberToSwitchOut int,
+    @partitionBoundaryToMerge datetime2(7)
+AS
+  BEGIN
+    
+    /* using XACT_ABORT to force a rollback on any error. */
+    SET XACT_ABORT ON;
+    
+    BEGIN TRANSACTION
+
+        /* Cleans up a staging table if there are existing rows. */
+        TRUNCATE TABLE dbo.ResourceChangeDataStaging;
+        
+        /* Switches a partition to the staging table. */
+        ALTER TABLE dbo.ResourceChangeData SWITCH PARTITION @partitionNumberToSwitchOut TO dbo.ResourceChangeDataStaging;
+        
+        /* Merges range to move lower boundary one partition ahead. */
+        ALTER PARTITION FUNCTION PartitionFunction_ResourceChangeData_Timestamp() MERGE RANGE(@partitionBoundaryToMerge);
```

Comment:
Just for my understanding, is there a use case where `@partitionBoundaryToMerge`  will not equal the lowest boundary value, or are we passing in `@partitionBoundaryToMerge` because it makes more sense to determine what it is elsewhere?

Reply:
In our case, no, it should be always the lowest boundary value. The reason for adding parameters is to make the contract explicit to ensure only what passed gets deleted.

Reply:
Okay thanks, that's good to know - does it make sense to add some sort of validation and/or documentation (comments) for this?

Reply:
Since the stored procedure is parameterized, the procedure can be used for the general purpose. It is safe. For the health PaaS side, we only pass the lowest boundary value.

Reply:
Added comments on the stored procedure.